src/osd_field.c

```diff
@@ -0,0 +1,359 @@
+// SPDX-License-Identifier: GPL-2.0-only
+#include <assert.h>
+#include <wlr/util/log.h>
+#include "common/mem.h"
+#include "config/rcxml.h"
+#include "view.h"
+#include "workspaces.h"
+#include "labwc.h"
+#include "osd.h"
+
+/* includes '%', terminating 's' and NULL byte, 8 is enough for %-9999W */
```

Comment:
Because we replace the `W` with `s`?

Reply:
right, the comment should likely state `%-9999s` instead as the `W` never ends up in the fmt buffer.